scene/main/viewport.cpp

```diff
@@ -1001,7 +1001,14 @@ void Viewport::set_world_2d(const Ref<World2D> &p_world_2d) {
 	}
 
 	if (p_world_2d.is_valid()) {
+		bool do_propogate = true;
+		if (world_2d.is_valid() && is_inside_tree()) {
+			do_propogate = true;
+		}
```

Comment:
`do_propogate` is always true. I think it should be
```suggestion
		bool do_propogate = world_2d.is_valid() && is_inside_tree();
```